Deprecate and remove JMX Trans (#65)
strimzi/proposals: 043-deprecate-and-remove-jmxtrans.md

# Deprecate and remove JMX Trans

## Current situation

> [JMX Trans](https://github.com/jmxtrans/jmxtrans) is a tool which allows to collect data from JMX endpoints of Java applications and send them to other applications and services.

Reviewer: ```suggestion
[JMX Trans](https://github.com/jmxtrans/jmxtrans) is a tool which allows data collection from JMX endpoints of Java applications to send them to other applications and services.
```

        attributes:  ["Count"]
        outputs: ["standardOut"]
  # ...
```

## Issues

The JMX Trans tool seems to be stale.
The last release is from March 31st 2021 - so more than year and a half ago.
Because the last release is so old, there are also many CVEs in the different dependencies it uses.
In addition to that, it is falling behind in other aspects as well.
For example, while we are moving all our container images to Java 17, JMX Trans does not run on Java 17, so it needs to stick with Java 11.

## Proposal

Since the JMX Trans project does not seem to be developed anymore, we should first deprecate the JMX Trans support and if nothing changes, we should remove the support.

In the first phase - as part of Strimzi 0.33 (currently expected in the second part of December or early January) - we will:
* Deprecate the `.spec.jmxTrans` API in the Kafka custom resource.
* Update the docs to indicate that JMX Trans is deprecated.
* Announce the deprecation to the users as part of the Strimzi 0.33 communication (release notes, change log etc.).

In the second phase - as part of Strimzi 0.35 (currently expected to be release in March or April 2023) - we will:

> * The `.spec.jmxTrans` API in the Kafka custom resource has to remain.

Reviewer: ```suggestion
* Retain the `.spec.jmxTrans` API in the Kafka custom resource.
```

  But it will stay deprecated and will be ignored by the operator (a warning will be issued if it is present in the custom resource).
* The Strimzi 0.35 and later will check for existence of the JMX Trans resources and delete them if they would exist.
  The rest of the operator code related to JMX Trans will be removed.
* The container image for JMX Trans will be removed.

> * Remove JMX Trans form the docs.

Reviewer 2: ```suggestion
* Remove JMX Trans from the docs.
```

> * Remove JMX Trans form the docs.

Reviewer 2: Maybe I would remove the `we will` and keep just the list and then change the `Remove JMX Trans from the docs` to `The JMX Trans will be removed from the docs`
